### PR TITLE
Optimised Cloverleaf3D

### PR DESCRIPTION
**Pull request by s1913388**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cloverleaf3d/package.py [hunk 1]
```diff
@@ -6,59 +6,78 @@
...
-            targets.append('FLAGS_PGI=')
-            targets.append('CFLAGS_PGI=')
-        elif '%xl' in self.spec:
-            targets.append('COMPILER=XLF')
-            targets.append('FLAGS_XLF=')
-            targets.append('CFLAGS_XLF=')
+        if "%gcc" in self.spec:
+            targets.append("COMPILER=GNU")
+            targets.append("FLAGS_GNU=-g -O3 -march=native -funroll-loops")
+            targets.append("CFLAGS_GNU=-g -O3 -march=native -funroll-loops")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/cloverleaf3d/package.py]: Please don't add `-march=native`, the spack compiler wrappers will add the march flag for you already.

File: var/spack/repos/builtin/packages/cloverleaf3d/package.py [hunk 2]
```diff
@@ -6,59 +6,78 @@
...
+            targets.append("FLAGS_PGI=")
+            targets.append("CFLAGS_PGI=")
+        elif "%xl" in self.spec:
+            targets.append("COMPILER=XLF")
+            targets.append("FLAGS_XLF=")
+            targets.append("CFLAGS_XLF=")
+        elif "%arm" in self.spec:
+            targets.append("COMPILER=ARM")
+            targets.append("FLAGS_ARM=-g -O3 -mcpu=native -funroll-loops")
+            targets.append("CFLAGS_ARM=-g -O3 -mcpu=native -funroll-loops")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/cloverleaf3d/package.py]: Same as above.

File: var/spack/repos/builtin/packages/cloverleaf3d/package.py [hunk 3]
```diff
@@ -6,59 +6,78 @@
...
-    tags = ['proxy-app']
+    tags = ["proxy-app"]
 
-    version('1.0', sha256='78d591728c61bdfd6175b3930df7652e09ed04fbcd01b3fc86fb2aa0f237a8ef')
+    maintainers = ["s1913388"]
 
-    variant('openacc', default=False, description='Enable OpenACC Support')
+    version(
+        "1.0", sha256="78d591728c61bdfd6175b3930df7652e09ed04fbcd01b3fc86fb2aa0f237a8ef"
+    )
```
Comment (haampie) [inline var/spack/repos/builtin/packages/cloverleaf3d/package.py:23]: why?

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/cloverleaf3d/package.py (hunk 3 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/cloverleaf3d/package.py:23]: The reformatting is presumably done by black. We're currently working on integrating black into Spack's tests, but until then I generally ignore style changes as long as they pass the tests.

File: var/spack/repos/builtin/packages/cloverleaf3d/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/cloverleaf3d/package.py]: Ping @s1913388

**Review by haampie — APPROVED** [review]

**Pull request merged**